azurerm/internal/services/frontdoor/frontdoor_resource.go

```diff
@@ -443,27 +443,27 @@ func resourceFrontDoor() *schema.Resource {
 							Optional: true,
 							Default:  0,
 						},
-						"custom_https_provisioning_enabled": {
-							Type:       schema.TypeBool,
-							Optional:   true,
-							Computed:   true,
-							Deprecated: "Deprecated in favour of `azurerm_frontdoor_custom_https_configuration` resource",
-						},
+						// "custom_https_provisioning_enabled": {
+						// 	Type:       schema.TypeBool,
+						// 	Optional:   true,
+						// 	Computed:   true,
+						// 	Deprecated: "Deprecated in favour of `azurerm_frontdoor_custom_https_configuration` resource",
+						// },
 						"web_application_firewall_policy_link_id": {
 							Type:     schema.TypeString,
 							Optional: true,
 							// TODO: validation that this is a resource id
 						},
-						"custom_https_configuration": {
-							Type:       schema.TypeList,
-							Optional:   true,
-							Computed:   true,
-							MaxItems:   1,
-							Deprecated: "Deprecated in favour of `azurerm_frontdoor_custom_https_configuration` resource",
-							Elem: &schema.Resource{
-								Schema: schemaCustomHttpsConfiguration(),
-							},
-						},
+						// "custom_https_configuration": {
```

Comment:
we never fully dug into this but when we tried to use this resource we had an issue with the wrong certificate being picked, I suspect something to do with the ordering of the endpoints, so your PR(s) may fix this issue and allow us to upgrade to it.

If possible I would suggest maintaining it to allow people to upgrade in stages

Reply:
I'll need to dive into why separate resource breaks then, something is fishy there.